tests/pipeline/test_engine.py

```diff
@@ -510,6 +519,107 @@ def test_rolling_and_nonrolling(self):
                 ),
             )
 
+    def test_factor_with_single_output(self):
+        dates = self.dates[5:10]
+        assets = self.assets
+        num_dates = len(dates)
+        open = USEquityPricing.open
+        open_values = array([self.constants[open]] * num_dates, dtype=float)
+        engine = SimplePipelineEngine(
+            lambda column: self.loader, self.dates, self.asset_finder,
+        )
+
+        single_output = OpenPrice(outputs=['open'])
+        pipeline = Pipeline(columns={'open': single_output.open})
+
+        results = engine.run_pipeline(pipeline, dates[0], dates[-1])
+        output_results = results['open'].unstack()
+        output_expected = {asset: open_values for asset in assets}
+
+        assert_frame_equal(
+            output_results,
+            DataFrame(output_expected, index=dates, columns=assets),
+        )
+
+    def test_factor_with_multiple_outputs(self):
+        dates = self.dates[5:10]
+        assets = self.assets
+        asset_ids = self.asset_ids
+        constants = self.constants
+        num_dates = len(dates)
+        num_assets = len(assets)
+        open = USEquityPricing.open
+        close = USEquityPricing.close
+        engine = SimplePipelineEngine(
+            lambda column: self.loader, self.dates, self.asset_finder,
+        )
+
+        def create_expected_results(expected_value, mask):
+            expected_values = where(mask, expected_value, nan)
+            return DataFrame(expected_values, index=dates, columns=assets)
+
+        multiple_outputs = MultipleOutputs()
+        open_price, close_price = MultipleOutputs()
+
+        # Ensure that both methods of accessing our outputs return the same
+        # things.
+        self.assertIs(open_price, multiple_outputs.open)
+        self.assertIs(close_price, multiple_outputs.close)
+
+        pipeline = Pipeline(
+            columns={'open_price': open_price, 'close_price': close_price},
```

Comment:
What happens if I add `multiple_outputs` itself as a column.  If that works, we should probably have a test for the behavior.  If it doesn't, then we should fail immediately if you try to set it as an output.